src/app/content/components/Page/scrollTargetManager.ts

```diff
@@ -36,17 +36,28 @@ const scrollToTop = () => {
   window.scrollTo(0, 0);
 };
 
-const getScrollTarget = (container: HTMLElement | null, hash: string): HTMLElement | null => {
-  return container && typeof(window) !== 'undefined' && hash
-    ? container.querySelector(`[id="${hash.replace(/^#/, '')}"]`)
-    : null;
+const getScrollTarget = (
+  container: HTMLElement | null, { type, value }: ScrollTarget
+): HTMLElement | null => {
+  if (!container || typeof(window) === 'undefined') { return null; }
+
+  if (type === 'hash') {
+    return container.querySelector(`[id="${value.replace(/^#/, '')}"]`);
+  } else if (type === 'highlight') {
+    return container.querySelector(`[data-highlight-id="${value}"]`);
+  }
+
+  return null;
 };
 
-const scrollTargetManager = (container: HTMLElement) => (previous: ScrollTargetProp, current: ScrollTargetProp) => {
+const scrollTargetManager = (container: HTMLElement) => (
+  previous: ScrollTargetProp,
+  current: ScrollTargetProp
+) => {
   if (previous.page !== current.page) {
-    scrollToTargetOrTop(container, current.hash);
-  } else if (previous.hash !== current.hash) {
-    scrollToTarget(container, current.hash);
+    scrollToTargetOrTop(container, current.target);
+  } else if (current.target && (!previous.target || previous.target.value !== current.target.value)) {
```

Comment:
could you do something like `} else if (!isEqual(previous.target, current.target)) {`?